File: apps/weight-loss-consultant-mobile/lib/constants/enums.dart
import 'package:json_annotation/json_annotation.dart';

enum Role{
  customer,
  trainer,
  admin,
}

extension RoleExtenstion on Role {
  String get value {
    switch (this) {
      case Role.admin:
        return 'admin';
      case Role.customer:
        return 'customer';
      case Role.trainer:
        return 'trainer';
    }
  }

}

enum Gender{
  male,
  female,
}

extension GenderExtension on Gender {
  int get value {
    switch (this) {
      case Gender.female:
        return 2;
      case Gender.male:
        return 1;
    }
  }

}

enum AccountStatus{
  active,
  inactive,
  pending
}

extension AccountStatusExtension on AccountStatus {
  int get value {
    switch (this) {
      case AccountStatus.active:
        return 1;
      case AccountStatus.inactive:
        return 0;
      case AccountStatus.pending:
        return 2;
    }
  }

}

File: apps/common/constants/enums.ts
export const JWT_CONFIG = {
  expireTime: "6000s"
}

export enum Role {
  Trainer = 'trainer',
  Customer = 'customer',
  Admin = 'admin'
}

export const RESET_PASSWORD_TOKEN_EXPIRED_TIME = {
  MILLISECOND: 5 * 60 * 1000,
  SECOND: 5 * 60,
  MINUTE: 5,
}

export const ENV_FILE_PATH = (() : string => {
  if (process.cwd().includes("dist")){
    return "./assets/.env"
  }
  return "./.env"
})();

export enum AccountStatus {
  ACTIVE = 1,
  INACTIVE = 0,
  PENDING = 2
}

export enum CampaignStatus {
  ACTIVE = 1,
  INACTIVE = 0,
  PENDING = 2
}

export enum PackageStatus {
  ACTIVE = 1,
  INACTIVE = 0,
  PENDING = 2
}

export enum RoleEnum {
  Customer = 'customer',
  Trainer = 'trainer',
  Admin = 'admin'
}

export enum Order {
  Ascending = "ASC",
  Descending = "DESC"
}

export enum Gender {
  MALE = 1,
  FEMALE = 2
}
